src/test/regress/sql/alter_index.sql

```diff
@@ -0,0 +1,39 @@
+CREATE SCHEMA alterindex;
+
+SET search_path TO "alterindex";
+SET citus.next_shard_id TO 980000;
+SET client_min_messages TO WARNING;
+SET citus.shard_count TO 4;
+SET citus.shard_replication_factor TO 1;
+
+-- test alter index set statistics
+CREATE TABLE t1 (a int, b int);
+SELECT create_distributed_table('t1','a');
+CREATE INDEX test_idx on t1 ((a+b));
+ALTER INDEX test_idx ALTER COLUMN 1 SET STATISTICS 4646;
+ALTER INDEX test_idx ALTER COLUMN 1 SET STATISTICS -4646;
+ALTER INDEX test_idx ALTER COLUMN 3 SET STATISTICS 4646;
+
+-- test alter index set statistics before distribution
+CREATE TABLE t2 (a int, b int);
+CREATE INDEX test_idx2 on t1 ((a+b), (a-b), (a*b));
```

Comment:
I've been trying to understand why the most of the code was tagged as "not covered with tests" 😄 
I think you meant to write `t2` here.